src/components/PuzzleGrid/index.jsx

```diff
@@ -0,0 +1,127 @@
+      newTilesArray.push(newTile);
+    }
+
+    const randomizedTilesArray = shuffle(newTilesArray);
+    // Set the randomized array as generated tiles
+    setCurrentPuzzleState(randomizedTilesArray);
+  };
+
+  useEffect(() => {
+    tilesGenerator();
+  }, []);
+
+  useEffect(() => {
+    checkSolution();
+  }, [currentPuzzleState]);
+
+  const handleSelect = (clickedTileId) => {
+    // if selectedTile's value is not equal to null (meaning there a tile seleted already)
+    // then when another tile is clicked swap both tiles' places
+    // else set the selectedTile to the id of the clicked tile
+    if (selectedTile !== null) {
+      const prevId = selectedTile;
+      swapTiles(prevId, clickedTileId);
+    } else {
+      setSelectedTile(clickedTileId);
+      const clickedTileIndex = currentPuzzleState.findIndex(
+        (tile) => tile.id === clickedTileId
+      );
+      // set the select value of the tile to true
+      currentPuzzleState[clickedTileIndex] = {
+        ...currentPuzzleState[clickedTileIndex],
+        selected: true,
+      };
+    }
+  };
+
+  const swapTiles = (id0, id1) => {
+    let newTiles = [...currentPuzzleState];
+    let index0 = currentPuzzleState.findIndex((tile) => tile.id === id0);
+    let index1 = currentPuzzleState.findIndex((tile) => tile.id === id1);
+    let tile0 = { ...currentPuzzleState[index0] };
+    let tile1 = { ...currentPuzzleState[index1] };
+    newTiles[index0] = tile1;
+    newTiles[index1] = tile0;
+    newTiles.forEach((tile) => (tile.selected = false));
+    setCurrentPuzzleState(newTiles);
+    setSelectedTile(null);
+  };
+
+  const checkSolution = () => {
+    // tiles are saved in an array indexed from 0 to n (6) and displayed in that order
+    // first condition checks if tile's id is equal to its index in the array and returns false/true
+    // since initial acc value is true, second condition checks if prev value is true and keeps its value
+    const isCorrect = currentPuzzleState.reduce((acc, curr, index) => {
+      if (curr.id !== index || acc === false) {
+        return false;
+      } else {
+        return acc;
+      }
+    }, true);
+    setIsSolved(isCorrect);
+  };
+
+  return (
+    <Col className="puzzleGrid">
+      {isSolved ? (
+        <img
+          className="puzzleGrid__imageDisplayed"
+          src={puzzledImage}
+          alt="tile"
+        />
+      ) : (
+        currentPuzzleState.map((tile) => {
+          return (
+            <Row
+              key={"tileNumber-" + tile.id}
+              className={
+                tile.selected
+                  ? "selectedTile puzzleGrid__tile"
+                  : "puzzleGrid__tile"
+              }
+              onClick={() => handleSelect(tile.id)}
+            >
+              <img
+                className="puzzleGrid__puzzledImage"
+                src={puzzledImage}
+                alt="tile from the image"
+                style={{ top: `-${tile.top}%` }}
```

Comment:
Usually inline styling is not allowed. But this is short and variable dependent. @louisrli Is there a better option?

Reply:
Exactly, each tile will have it's own `top` value so I did it here. please let me know if there is an alternative

Reply:
you might be able to use n-th child but i'd probably prefer this inline styling here since its dynamically calculated